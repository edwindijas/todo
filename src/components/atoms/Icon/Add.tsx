import React from 'react'
import ReactComponent from '@material-design-icons/svg/round/add.svg?react'
import Icon from '.'

const AddIcon = (props: React.SVGAttributes<SVGElement>): JSX.Element => (
  <Icon {...props} component={ReactComponent} />
)

export default AddIcon
