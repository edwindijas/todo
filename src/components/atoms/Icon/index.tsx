import propTypes, { type InferProps } from 'prop-types'

export default function Icon ({
  component: Component,
  className: importClassName
}: InferProps<typeof Icon.propTypes>): JSX.Element {
  const className = `${importClassName} w-full h-full`
  return <Component
    className={className}
  />
}

Icon.propTypes = {
  component: propTypes.func.isRequired,
  className: propTypes.string
}

Icon.defaultProps = {
  className: ''
}

