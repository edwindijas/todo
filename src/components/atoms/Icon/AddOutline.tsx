import React from 'react'
import ReactComponent from '@material-design-icons/svg/outlined/add.svg?react'
import Icon from '.'

const AddOutlineIcon = (props: React.SVGAttributes<SVGElement>): JSX.Element => (
  <Icon {...props} component={ReactComponent} />
)

export default AddOutlineIcon
