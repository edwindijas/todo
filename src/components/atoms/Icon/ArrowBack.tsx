import React from 'react'
import ReactComponent from '@material-design-icons/svg/round/arrow_back.svg?react'
import Icon from '.'

const ArrowBackIcon = (props: React.SVGAttributes<SVGElement>): JSX.Element => (
  <Icon {...props} component={ReactComponent} />
)

export default ArrowBackIcon
