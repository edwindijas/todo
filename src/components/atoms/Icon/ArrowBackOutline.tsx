import React from 'react'
import ReactComponent from '@material-design-icons/svg/outlined/arrow_back.svg?react'
import Icon from '.'

const ArrowBackOutlineIcon = (props: React.SVGAttributes<SVGElement>): JSX.Element => (
  <Icon {...props} component={ReactComponent} />
)

export default ArrowBackOutlineIcon
