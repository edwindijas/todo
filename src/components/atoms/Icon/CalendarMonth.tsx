import React from 'react'
import ReactComponent from '@material-design-icons/svg/round/calendar_month.svg?react'
import Icon from '.'

const CalendarMonthIcon = (props: React.SVGAttributes<SVGElement>): JSX.Element => (
  <Icon {...props} component={ReactComponent} />
)

export default CalendarMonthIcon
