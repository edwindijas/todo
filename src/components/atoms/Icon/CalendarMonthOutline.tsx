import React from 'react'
import ReactComponent from '@material-design-icons/svg/outlined/calendar_month.svg?react'
import Icon from '.'

const CalendarMonthOutlineIcon = (props: React.SVGAttributes<SVGElement>): JSX.Element => (
  <Icon {...props} component={ReactComponent} />
)

export default CalendarMonthOutlineIcon
