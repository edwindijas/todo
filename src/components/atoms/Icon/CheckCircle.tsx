import React from 'react'
import ReactComponent from '@material-design-icons/svg/round/check_circle.svg?react'
import Icon from '.'

const CheckCircleIcon = (props: React.SVGAttributes<SVGElement>): JSX.Element => (
  <Icon {...props} component={ReactComponent} />
)

export default CheckCircleIcon
