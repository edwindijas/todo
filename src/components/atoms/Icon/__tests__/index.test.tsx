import { render } from '@testing-library/react'
import PropTypes, {type InferProps} from 'prop-types'
import Icon from '..'


function SampleComponent ({className}: InferProps<typeof SampleComponent.propTypes>): JSX.Element {
  return <div className={className ?? ''} >Hello World</div>
} 

SampleComponent.propTypes = {
  className: PropTypes.string
}

SampleComponent.defaultProps = {
  className: ''
} 

describe ('the main icon renders correctly', () => {
  it ('matches the snapshot', () => {
    const {container} = render(<Icon className='bg-black h-34' component={SampleComponent} />)
    expect(container).toMatchSnapshot()
  })
})