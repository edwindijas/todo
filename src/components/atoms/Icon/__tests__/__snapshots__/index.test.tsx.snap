// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the main icon renders correctly matches the snapshot 1`] = `
<div>
  <div
    class="bg-black h-34 w-full h-full"
  >
    Hello World
  </div>
</div>
`;
